src/common/xo/migrate-vm-modal/index.js

```diff
@@ -163,6 +164,33 @@ export default class MigrateVmModalBody extends BaseComponent {
 
     // Intra-pool
     const defaultSr = pools[host.$pool].default_SR
+
+    let targetSr
+    if (
+      isSrShared(this._getObject(defaultSr)) ||
+      some(host.$PBDs, pbd => this._getObject(pbd).SR === defaultSr)
+    ) {
+      targetSr = defaultSr
+    } else {
+      for (let pbd of host.$PBDs) {
+        const sr = this._getObject(this._getObject(pbd).SR)
```

Comment:
You can create a `getSrsByHost` in the `connectStore` and get the SRs list easily instead of this.